core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -903,21 +903,31 @@ public void deleteTenant(String tenantUniqueID) throws UserStoreException {
             clearTenantCaches(tenant);
             invalidateCacheManager(tenant.getDomain());
         }
-        String sqlStmt = TenantConstants.DELETE_TENANT_BY_UUID_SQL;
-        try (Connection dbConnection = getDBConnection()) {
-            try (PreparedStatement prepStmt = dbConnection.prepareStatement(sqlStmt)) {
+        String tenantDeleteSqlStmt = TenantConstants.DELETE_TENANT_BY_UUID_SQL;
+        String tenantDomainDeleteSqlStmt = TenantConstants.DELETE_TENANT_DOMAIN_BY_TENANT_ID_SQL;
+        Connection dbConnection = null;
+
+        try {
+            dbConnection = getDBConnection();
+            try (PreparedStatement prepStmt = dbConnection.prepareStatement(tenantDeleteSqlStmt)) {
```

Comment:
Let's add comments here if we are unable to make methods for each functionality

Reply:
Added separate methods
fixed 775da9a1e009be12698bc9ac9c86a8491b975b16